messaging/messaging.go

```diff
@@ -209,42 +205,328 @@ func (a *AndroidConfig) UnmarshalJSON(b []byte) error {
+	clr, err := newColor(l.Color)
+	if err != nil {
+		return nil, err
+	}
+
+	temp := struct {
+		Color            *color `json:"color"`
+		LightOnDuration  string `json:"light_on_duration"`
+		LightOffDuration string `json:"light_off_duration"`
+	}{
+		Color:            clr,
+		LightOnDuration:  durationToString(time.Duration(l.LightOnDurationMillis) * time.Millisecond),
+		LightOffDuration: durationToString(time.Duration(l.LightOffDurationMillis) * time.Millisecond),
+	}
+	return json.Marshal(temp)
+}
+
+// UnmarshalJSON unmarshals a JSON string into an LightSettings (for internal use only).
+func (l *LightSettings) UnmarshalJSON(b []byte) error {
+	temp := struct {
+		Color            *color `json:"color"`
+		LightOnDuration  string `json:"light_on_duration"`
+		LightOffDuration string `json:"light_off_duration"`
+	}{}
+	if err := json.Unmarshal(b, &temp); err != nil {
+		return err
+	}
+
+	on, err := stringToDuration(temp.LightOnDuration)
+	if err != nil {
+		return err
+	}
+
+	off, err := stringToDuration(temp.LightOffDuration)
+	if err != nil {
+		return err
+	}
+
+	l.Color = temp.Color.toString()
+	l.LightOnDurationMillis = int64(on / time.Millisecond)
+	l.LightOffDurationMillis = int64(off / time.Millisecond)
+	return nil
+}
+
+func durationToString(ms time.Duration) string {
+	seconds := int64(ms / time.Second)
+	nanos := int64((ms - time.Duration(seconds)*time.Second) / time.Nanosecond)
+	if nanos > 0 {
+		return fmt.Sprintf("%d.%09ds", seconds, nanos)
+	}
+	return fmt.Sprintf("%ds", seconds)
+}
+
+func stringToDuration(s string) (time.Duration, error) {
+	segments := strings.Split(strings.TrimSuffix(s, "s"), ".")
+	if len(segments) != 1 && len(segments) != 2 {
+		return 0, fmt.Errorf("incorrect number of segments in ttl: %q", s)
+	}
+
+	seconds, err := strconv.ParseInt(segments[0], 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse %s: %v", s, err)
+	}
+
+	ttl := time.Duration(seconds) * time.Second
+	if len(segments) == 2 {
+		nanos, err := strconv.ParseInt(strings.TrimLeft(segments[1], "0"), 10, 64)
+		if err != nil {
+			return 0, fmt.Errorf("failed to parse %s: %v", s, err)
+		}
+		ttl += time.Duration(nanos) * time.Nanosecond
+	}
+
+	return ttl, nil
+}
+
+type color struct {
+	Red   float64 `json:"red"`
+	Green float64 `json:"green"`
+	Blue  float64 `json:"blue"`
+	Alpha float64 `json:"alpha"`
+}
+
+func newColor(clr string) (*color, error) {
+	red, err := strconv.ParseInt(clr[1:3], 16, 32)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %v", clr, err)
+	}
+
+	green, err := strconv.ParseInt(clr[3:5], 16, 32)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %v", clr, err)
+	}
+
+	blue, err := strconv.ParseInt(clr[5:7], 16, 32)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse %s: %v", clr, err)
+	}
+
+	return &color{
+		Red:   float64(red) / 255.0,
+		Green: float64(green) / 255.0,
+		Blue:  float64(blue) / 255.0,
+		Alpha: 1.0,
```

Comment:
I think you can use the `Alpha` channel to control the brightness of the LED. Should we support it in the API?
We could add it in a future PR, too

Reply:
Added